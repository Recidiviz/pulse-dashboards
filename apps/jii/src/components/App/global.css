/* styled-components discourages @import for global styles so this is separate */
@import "modern-normalize"