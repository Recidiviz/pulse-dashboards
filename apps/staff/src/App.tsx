// Recidiviz - a data platform for criminal justice reform
// Copyright (C) 2021 Recidiviz, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
// =============================================================================

import "react-aspect-ratio/aspect-ratio.css";
import "./assets/scripts/index";
import "./assets/styles/index.scss";

import { useEffect } from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import tk from "timekeeper";
import { QueryParamProvider } from "use-query-params";
import { ReactRouter6Adapter } from "use-query-params/adapters/react-router-6";

import NotFound from "./components/NotFound";
import SentryErrorBoundary from "./components/SentryErrorBoundary";
import StoreProvider from "./components/StoreProvider";
import StyledToaster from "./components/StyledToaster";
import VerificationNeeded from "./components/VerificationNeeded";
import ProtectedLayout from "./ProtectedLayout";
import { initI18n } from "./utils/i18nSettings";
import initIntercomSettings from "./utils/initIntercomSettings";
import { isDemoMode } from "./utils/isDemoMode";

if (!isDemoMode()) {
  initIntercomSettings();
}
initI18n();

function App() {
  useEffect(() => {
    if (isDemoMode()) {
      const demoDate = new Date("2022-03-22T10:30:00");
      tk.travel(demoDate); // Freeze time to the desired date and time
    }

    return () => {
      if (isDemoMode()) {
        tk.reset(); // Reset the time to the actual system time when the component is unmounted
      }
    };
  }, []);
  return (
    <StoreProvider>
      <Router>
        <SentryErrorBoundary>
          <QueryParamProvider adapter={ReactRouter6Adapter}>
            <Routes>
              <Route path="/verify" element={<VerificationNeeded />} />
              <Route path="/*" element={<ProtectedLayout />} />
              <Route path="*" element={<NotFound />} />
            </Routes>
            <StyledToaster />
          </QueryParamProvider>
        </SentryErrorBoundary>
      </Router>
    </StoreProvider>
  );
}

export default App;
