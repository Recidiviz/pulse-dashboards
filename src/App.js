import React, { useEffect, useState } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

import * as $ from 'jquery';
import Masonry from 'masonry-layout';

import PrivateRoute from "./components/PrivateRoute";
import Loading from "./components/Loading";
import SideBar from "./components/SideBar";
import TopBar from "./components/TopBar";
import Footer from "./components/Footer";
import Home from "./views/Home";
import NotFound from "./views/NotFound";
import Profile from "./views/Profile";
import Revocations from "./views/Revocations";
import Reincarcerations from "./views/Reincarcerations";
import Snapshots from "./views/Snapshots";
import ExternalApi from "./views/ExternalApi";
import { useAuth0 } from "./react-auth0-spa";
import "./assets/scripts/index.js";

// styles
import "./assets/styles/index.scss";

// fontawesome
import initFontAwesome from "./utils/initFontAwesome";
initFontAwesome();

const App = () => {
  const [sideBarCollapsed, setSideBarCollapsed] = useState("");
  const { loading, isAuthenticated } = useAuth0();

  function toggleCollapsed() {
    const currentlyCollapsed = sideBarCollapsed === "is-collapsed";
    if (currentlyCollapsed) {
      setSideBarCollapsed("");
    } else {
      setSideBarCollapsed("is-collapsed");
    }
  }

  // TODO: Replace this jQuery with actual React Masonry and toggle components
  useEffect(() => {
    new Masonry('.masonry', {
      itemSelector: '.masonry-item',
      columnWidth: '.masonry-sizer',
      percentPosition: true,
    });

    // Reinitialize masonry inside each panel after a collapsible accordion is expanded or collapsed
    $('.accordion').each(function () {
      const $this = $(this);

      $this.on('shown.bs.collapse hidden.bs.collapse', () => {
        if ($('.masonry').length > 0) {
          new Masonry('.masonry', {
            itemSelector: '.masonry-item',
            columnWidth: '.masonry-sizer',
            percentPosition: true,
          });
        }
      });
    });

    // ٍSidebar Toggle
    $('.sidebar-toggle').on('click', (e) => {
      toggleCollapsed();
      e.preventDefault();
    });

    /**
     * Wait untill sidebar fully toggled (animated in/out)
     * then trigger window resize event in order to recalculate
     * masonry layout widths and gutters.
     */
    $('#sidebar-toggle').click((e) => {
      e.preventDefault();
      setTimeout(() => {
        window.dispatchEvent(window.EVENT);
      }, 300);
    });
  });

  if (loading) {
    return <Loading />;
  }

  let containerClass = "wide-page-container";
  if (isAuthenticated) {
    containerClass = "page-container";
  }

  return (
    <Router>
      <div id="app" className={sideBarCollapsed}>
        <div>
          <meta charSet="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
          <title>North Dakota</title>
          <div>
            {isAuthenticated && (
            <SideBar />
            )}
            <div className={containerClass}>
              <TopBar pathname={window.location.pathname} />
              <Switch>
                <Route path="/" exact component={Home} />
                <PrivateRoute path="/snapshots" component={Snapshots} />
                <PrivateRoute path="/revocations" component={Revocations} />
                <PrivateRoute path="/reincarcerations" component={Reincarcerations} />
                <PrivateRoute path="/profile" component={Profile} />
                <PrivateRoute path="/external-api" component={ExternalApi} />
                <Route component={NotFound} />
              </Switch>
              <Footer />
            </div>
          </div>
        </div>
      </div>
    </Router>
  );
};

export default App;
