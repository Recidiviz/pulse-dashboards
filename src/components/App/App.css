/*
 * Recidiviz - a platform for tracking granular criminal justice metrics in real time
 * Copyright (C) 2018 Recidiviz, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 * ============================================================================
*/

.app-container {
	background: #d6d4d3;
	font-family: 'Nunito', monospace;
	color: #766e6e;
	min-height: 100vh;
}

a {
	text-decoration: none;
	color: rgb(85, 26, 139);
	outline: none;
}

a:hover, a:focus {
	color: #c82222;
}

.bg-alert-blue {
 background-color: #eeeeee;
}

.bg-our-gray {
  background-color: #d6d4d3;
}

.text-our-gray {
	color: #766e6e;
}

.text-our-highlight {
	color: #c82222;
}
