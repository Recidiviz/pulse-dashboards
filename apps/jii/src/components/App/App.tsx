// Recidiviz - a data platform for criminal justice reform
// Copyright (C) 2024 Recidiviz, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
// =============================================================================

import "./global.css";
import "@fontsource/public-sans/400.css";
import "@fontsource/public-sans/500.css";
import "@fontsource/public-sans/600.css";
import "@fontsource/public-sans/700.css";
import "@fontsource/libre-baskerville";

import {
  GlobalStyle as GlobalStyleBase,
  palette,
  typography,
} from "@recidiviz/design-system";
import { ErrorBoundary, withSentryReactRouterV6Routing } from "@sentry/react";
import { Route, Routes } from "react-router-dom";
import styled, { createGlobalStyle } from "styled-components/macro";

import { initializeSentry } from "../../apis/Sentry/initializeSentry";
import { ErrorPage } from "../ErrorPage/ErrorPage";
import { NotFound } from "../NotFound/NotFound";
import { PageAfterLogin } from "../pages/PageAfterLogin";
import { PageEligibility } from "../pages/PageEligibility";
import { PageEligibilityHome } from "../pages/PageEligibilityHome";
import { PageHome } from "../pages/PageHome";
import { PageLanding } from "../pages/PageLanding";
import { PageOpportunityEligibility } from "../pages/PageOpportunityEligibility";
import { PageRoot } from "../pages/PageRoot";
import { PageSearch } from "../pages/PageSearch";
import { PageStateLanding } from "../pages/PageStateLanding";
import { PageVerifyEmail } from "../pages/PageVerifyEmail";
import { StaticPage } from "../StaticPage/StaticPage";
import { StoreProvider } from "../StoreProvider/StoreProvider";

const StyledApp = styled.div``;

const GlobalStyle = createGlobalStyle`
  body {
    ${typography.Sans16}
    
    background: ${palette.white};
    color: ${palette.pine2};
  }
`;

initializeSentry();

// doing this once (at the root off all <Routes>) allows Sentry to trace client-side URLs
const SentryRoutes = withSentryReactRouterV6Routing(Routes);

export function App() {
  return (
    <ErrorBoundary showDialog fallback={ErrorPage}>
      <StoreProvider>
        <GlobalStyleBase />
        <GlobalStyle />

        <StyledApp>
          <SentryRoutes>
            <Route path="/" element={<PageRoot />}>
              <Route index element={<PageHome />} />
              <Route path="welcome" element={<PageLanding />} />
              <Route path=":landingPageUrl" element={<PageStateLanding />} />
              <Route path="verify" element={<PageVerifyEmail />} />
              <Route path="after-login" element={<PageAfterLogin />} />
              <Route path="eligibility" element={<PageEligibility />}>
                <Route index element={<PageEligibilityHome />} />
                <Route path="search" element={<PageSearch />} />
                <Route path=":opportunityUrl">
                  <Route index element={<PageOpportunityEligibility />} />
                  <Route path="about" element={<StaticPage pageId="about" />} />
                  <Route
                    path="requirements"
                    element={<StaticPage pageId="requirements" />}
                  />
                  <Route
                    path="next-steps"
                    element={<StaticPage pageId="nextSteps" />}
                  />
                </Route>
              </Route>
              <Route path="*" element={<NotFound />} />
            </Route>
          </SentryRoutes>
        </StyledApp>
      </StoreProvider>
    </ErrorBoundary>
  );
}
