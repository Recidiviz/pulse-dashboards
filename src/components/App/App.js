/*
 * Recidiviz - a platform for tracking granular criminal justice metrics in real time
 * Copyright (C) 2018 Recidiviz, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 * ============================================================================
*/

import React from 'react';

import ChartsContainer from '../ChartsContainer';
import Header from '../Header';
import Footer from '../Footer';
import KPIsContainer from '../KPIsContainer';
import Sidebar from '../Sidebar';

import './App.css';

import Organization from '../../config/constants';

function App() {
  return (
    <div className="app-container">
      <Sidebar />
      <Header
        title={Organization.name}
        contactEmail={Organization.primaryContact}
      />
      <KPIsContainer />
      <ChartsContainer />
      <Footer />
    </div>
  );
}

export default App;
