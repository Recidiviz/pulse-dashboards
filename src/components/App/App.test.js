/*
 * Recidiviz - a platform for tracking granular criminal justice metrics in real time
 * Copyright (C) 2018 Recidiviz, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 * ============================================================================
*/

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

describe('<App />', () => {
  test('renders a <Header /> component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('Header').exists()).toBe(true);
  });

  test('renders a <Footer /> component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('Footer').exists()).toBe(true);
  });

  test('matches saved snapshot', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
