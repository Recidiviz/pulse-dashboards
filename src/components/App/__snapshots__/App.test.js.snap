// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches saved snapshot 1`] = `
<div
  className="app-container"
>
  <Sidebar />
  <Header
    contactEmail="team@recidiviz.com"
    title="Recidiviz"
  />
  <KPIsContainer />
  <ChartsContainer />
  <Footer />
</div>
`;
